rn/Teacher/src/modules/submissions/list/get-submissions-props.js

```diff
@@ -142,45 +128,85 @@ export function pendingProp (assignmentContent?: AssignmentContentState, courseC
   return assignmentContent.submissions.pending > 0 || courseContent.enrollments.pending > 0
 }
 
+// If a submission does not have a grade or an actual submission it will not have a group attached to the submission.
+// So if it's there we get the group from entities, but if not then we will look up the group in entities based on
+// the category id and if the user is in the group.
+export function getGroup (groups: GroupsState, submission: SubmissionWithHistory, categoryID: ?string): ?Group {
+  if (submission.group && submission.group.id != null) {
+    let group = groups[submission.group.id]
+    if (!group) return
+    return group.group
+  }
+
+  if (!categoryID) return
+
+  // $FlowFixMe
+  let groupValues: Array<{ group: Group }> = Object.values(groups)
+  let group = groupValues
+    .filter(({ group }) => group && group.group_category_id === categoryID)
+    .find(({ group }) => group.users && group.users.map(({ id }) => id).includes(submission.user_id))
+
+  if (!group) return
+  return group.group
+}
+
+export function groupPropsForSubmissionsAndDueDate (submission: SubmissionWithHistory, dueDate: ?string, group: ?Group) {
+  if (!group) return
+  return {
+    // just a precaution... we're filtering out empty groups below
+    userID: group.users && group.users.length > 0 ? group.users[0].id : 'none',
```

Comment:
How do we know we can grab the first user out of `group.users`? I'm not following this part.

Reply:
if groups.users length is greater than 0 than there is guaranteed to have at least one user.